proposals/0652-import-shadowing.rst

```diff
@@ -258,43 +258,70 @@ something like
 
  module A (foo) where
 
- import M -- This exports "foo"
+ import M (foo)
 
  foo = ...
 
 then the ``foo`` exported by ``A`` should be the one defined in
 ``A``'s top-level.
 
-When modules are reexported wholesale, shadowing doesn't come into
-play, and so we keep the behaviour without this extension: the form
-``module M`` names the set of all entities that are in scope with both
-an unqualified name ``e`` and a qualified name ``M.e``. Example:
+.. _module-reexports:
+
+Module reexports
+^^^^^^^^^^^^^^^^
+
+In an export list, `Paragraph 5 of Section 5.2 of the Haskell 2010 report
+<https://www.haskell.org/onlinereport/haskell2010/haskellch5.html#x11-1000005.2>`_
+specifies that the form ``module M`` names the set of all entities
+that are in scope with both an unqualified name ``e`` and a qualified
+name ``M.e``.  So in this example, *without* ``ImportShadowing``:
```

Comment:
Presentation point.  The very-important "without ImportShadowing" appears before the example, while the consequence appears after.  Easy to mis-read.  I suggest
```
Consider this example:
    module A( module M ) where ...
   ..etc

Without ImportShadowing the `module M` exports...

With ImportShadowing, however, ....
```